features/steps/steps.rb

```diff
@@ -0,0 +1,69 @@
+Then("the Content-Type header is valid multipart form-data") do
+  expected = /^multipart\/form-data; boundary=--------------------------\d+$/
+  actual = Server.current_request[:request]["content-type"]
+
+  assert_match(expected, actual)
+end
+
+def read_expected_file(fixture, file_name)
```

Comment:
I'd expect to (with a bit of tweaking for the multipart side of things) use a similar step to [this one](https://github.com/bugsnag/maze-runner/blob/master/lib/features/steps/request_assertion_steps.rb#L156) to make this comparison.

New steps:
- Then the multipart payload field {string} matches the JSON fixture in {string}
- Then the multipart payload field {string} matches the fixture in {string}

Reply:
This isn't a step/assertion, it exists for passing the form data as a string to `assert_payload_fields_match`

`value_compare` doesn't pass if this isn't explicitly converted to a string because it checks the types of both arguments before comparing anything and fails if they are different

I assume the step you linked works because assert_equal must silently call `to_s` somewhere

Reply:
Quite possibly.  I'll break the multipart steps out of the request-assertion steps and see if this can be sorted.